File: Server/Server/Data/Models/Topic.cs
namespace Server.Data.Models
{
    public class Topic
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int ForumId { get; set; }
        public Forum Forum { get; set; }
        public List<Post> Posts { get; set; }
        public string UserId { get; set; }
        public User User { get; set; }
    }
}

File: ServerLaravel/app/Models/Topic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Topic extends Model
{
    use HasFactory;
    protected $table = "topics";
    protected $fillable = [
        "name",
        "description",
        "forum_id",
        "user_id"
    ];

    public function posts() {
        return $this->hasMany(Post::class, "topic_id", "id");
    }

    public function user() {
        return $this->hasOne(User::class, "id", "user_id");
    }
    
}
